### PR TITLE
Clean-up hardcoded vscode messages names

### LINKED ISSUE #68: Clean-up hardcoded vscode messages names
The message name of messages sent between webviews and extension that use the vscode API `vscode.postMessage()` should not be hard-coded.

### PR DESCRIPTION
**Pull request by hriday-panchasara**
Fixes #68

### CODE REVIEW

**Comment by hriday-panchasara** [comment]
> This PR doesn't build successfully for me locally (running `yarn` a few consecutive times), after rebasing it on latest master commit [b603bb0](https://github.com/eclipse-cdt-cloud/vscode-trace-extension/commit/b603bb08735a056a2d526649c227f9b0ff316e41). Can you rebase it in here too, so we may see what CI says?

Rebased on latest master and it builds for me, could you check and see if build error's still exist?

**Comment by marco-miller** [comment]
> > This PR doesn't build successfully for me locally (running `yarn` a few consecutive times), after rebasing it on latest master commit [b603bb0](https://github.com/eclipse-cdt-cloud/vscode-trace-extension/commit/b603bb08735a056a2d526649c227f9b0ff316e41). Can you rebase it in here too, so we may see what CI says?
> 
> Rebased on latest master and it builds for me, could you check and see if build error's still exist?

Yes, it builds the first time locally and with CI herein. However, running `yarn` more than just once leads to these build errors below, not there with master commit b603bb0. More notes follow down below.

```
ERROR in ./src/trace-explorer/available-views/trace-explorer-available-views-webview-provider.ts 7:0-90
Module not found: Error: Can't resolve 'vscode-trace-webviews/lib/common/vscode-message-manager' in '/home/user/git/vscode-trace-extension/vscode-trace-extension/src/trace-explorer/available-views'
 @ ./src/extension.ts 5:0-135 12:35-70 13:73-117
ERROR in ./src/trace-explorer/opened-traces/trace-explorer-opened-traces-webview-provider.ts 7:0-90
Module not found: Error: Can't resolve 'vscode-trace-webviews/lib/common/vscode-message-manager' in '/home/user/git/vscode-trace-extension/vscode-trace-extension/src/trace-explorer/opened-traces'
 @ ./src/extension.ts 6:0-133 10:31-68 11:73-119
ERROR in ./src/trace-viewer-panel/trace-viewer-webview-panel.ts 5:0-90
Module not found: Error: Can't resolve 'vscode-trace-webviews/lib/common/vscode-message-manager' in '/home/user/git/vscode-trace-extension/vscode-trace-extension/src/trace-viewer-panel'
 @ ./src/trace-explorer/analysis-tree.ts 3:0-84 44:4-39
 @ ./src/extension.ts 3:0-66 21:33-49
3 errors have detailed information that is not shown.
Use 'stats.errorDetails: true' resp. '--stats-error-details' to show it.
webpack 5.76.0 compiled with 3 errors in 1943 ms
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
lerna ERR! yarn run prepare exited 1 in 'vscode-trace-extension'
```

- Above, I removed a few redundant lines and parts from the original log.
- I'm unable to rebuild even if removing all `node_modules` locally.
- Using yarn --version `1.22.19` locally; along with nvm `v16.19.1`, the previous LTS version (now 16.20).

**Review by marco-miller — CHANGES_REQUESTED** [review]
(Per above.) -Please let me know how running `yarn` multiple times goes for you @hriday-panchasara; thanks. If need be, we may ask another Reviewer's own local build trials to compare.

**Comment by hriday-panchasara** [comment]
> (Per above.) -Please let me know how running `yarn` multiple times goes for you @hriday-panchasara; thanks. If need be, we may ask another Reviewer's own local build trials to compare.

Yup, its failing for me. Not sure why, the IDE does not throw any warnings and the package.json of `vscode-trace-extension` does list `vscode-trace-webviews` as a dependency. Its something related to the configuration in pacakge.json/tsconfig.json of one of these pacakages, but I can't wrap my head around what the exact issue is

**Comment by marco-miller** [comment]
> > (Per above.) -Please let me know how running `yarn` multiple times goes for you @hriday-panchasara; thanks. If need be, we may ask another Reviewer's own local build trials to compare.
> 
> Yup, its failing for me. Not sure why, the IDE does not throw any warnings and the package.json of `vscode-trace-extension` does list `vscode-trace-webviews` as a dependency. Its something related to the configuration in pacakge.json/tsconfig.json of one of these pacakages, but I can't wrap my head around what the exact issue is

You could try starting this change all over again, only applying one single code change at a time; to confirm `yarn && yarn` each time. This might help you spot which exact diff happens to break the rebuild, thus knowing how to fix that maybe.

**Comment by bhufmann** [comment]
I'm able to reproduce the issues after the 2nd build. 

Just for clarification, each webview runs an own application that is bundle using webpack. During the build the webpack output of vscode-trace-webview is stored in the vscode-trace-extension/pack directory and the vscode-trace-extension bundles it into the VsCode extension. So, the dependency in vsocde-trace-extension\package.json is not really needed. I suspect it's there to make sure that the webpack outputs are built first. To be able to re-use source code directly like the message definitions in this PR, I guess the build is not setup properly. However, I can't pin-point to what to change at the moment. We could try to re-structure the code so that share code are in a separate package, e.g. vscode-trace-common and try to hook it up. Alternatively, the vscode-trace-webviews could be moved into vscode-trace-extension directory and the build process updated accordingly. These are just suggestions and without trying it I can't say for sure if they work.

**Comment by hriday-panchasara** [comment]
> I'm able to reproduce the issues after the 2nd build.
> 
> Just for clarification, each webview runs an own application that is bundle using webpack. During the build the webpack output of `vscode-trace-webview` is stored in the `vscode-trace-extension/pack` directory and the `vscode-trace-extension` bundles it into the VsCode extension. So, the dependency in `vsocde-trace-extension\package.json` is not really needed. I suspect it's there to make sure that the webpack outputs are built first.
> 
> To be able to re-use source code directly like the message definitions in this PR, I guess the dependencies are not setup properly. However, I can't pin-point to what to change to fix it at the moment. We could also try to re-structure the code so that shared code is in a separate package, e.g. `vscode-trace-common` and try to hook it up. Alternatively, the `vscode-trace-webviews` directory could be moved into `vscode-trace-extension` directory and the build process updated accordingly. These are just suggestions and without trying it I can't say for sure if they work.

Thanks for the suggestions Bernd. I have restructured the code and put the `vscode-message-manager.ts` in a separate package called `vscode-trace-common`. It builds successfully for me locally. Let me know if this is acceptable

File: vscode-trace-common/src/messages/vscode-message-manager.ts [hunk 1]
```diff
@@ -20,6 +20,31 @@ export interface VsCodeTraceAction {
...
+    EXPERIMENT_SELECTED: 'experimentSelected',
+    NEW_STATUS: 'newStatus',
+    OPENED_TRACES_UPDATED: 'openedTracesUpdated',
+    OPEN_OVERVIEW: 'open-overview',
+    OUTPUT_ADDED: 'outputAdded',
+    REFACTOR: 'refactor',
+    RE_OPEN_TRACE: 'reopenTrace',
+    REMOVE_STATUS: 'rmStatus',
+    RESET_ZOOM: 'reset-zoom',
+    SAVE_AS_CVS: 'saveAsCsv',
```
Comment (bhufmann) [inline vscode-trace-common/src/messages/vscode-message-manager.ts:39]: should be SAVE_AS_CSV

File: vscode-trace-common/src/messages/vscode-message-manager.ts [hunk 2]
```diff
@@ -20,6 +20,31 @@ export interface VsCodeTraceAction {
...
+    SET_EXPERIMENT: 'set-experiment',
+    SET_THEME: 'set-theme',
+    SET_TSP_CLIENT: 'set-tspClient',
+    TRACE_VIEWER_TAB_ACTIVATED: 'traceViewerTabActivated',
+    UPDATE_PROPERTIES: 'updateProperties',
+    WEBVIEW_READY: 'webviewReady',
+};
+
 export class VsCodeMessageManager extends Messages.MessageManager {
```
Comment (bhufmann) [inline vscode-trace-common/src/messages/vscode-message-manager.ts:48]: the package.json is missing dependency to traceviewer-base

**Review by bhufmann — CHANGES_REQUESTED** [review]

**Review by bhufmann — COMMENTED** [review]
LGTM. Since I changed some parts @marco-miller could you please review this one too? Thanks.

File: vscode-trace-common/src/messages/vscode-message-manager.ts [hunk 3]
```diff
@@ -20,6 +20,35 @@ export interface VsCodeTraceAction {
...
+    EXPERIMENT_SELECTED: 'experimentSelected',
+    NEW_STATUS: 'newStatus',
+    OPENED_TRACES_UPDATED: 'openedTracesUpdated',
+    OPEN_OVERVIEW: 'open-overview',
+    OUTPUT_ADDED: 'outputAdded',
+    REFACTOR: 'refactor',
+    RE_OPEN_TRACE: 'reopenTrace',
+    REMOVE_STATUS: 'rmStatus',
+    RESET_ZOOM: 'reset-zoom',
+<<<<<<< HEAD
```
Comment (bhufmann) [inline vscode-trace-common/src/messages/vscode-message-manager.ts:39]: my mistake

**Comment by hriday-panchasara** [comment]
> LGTM. Since I changed some parts @marco-miller could you please review this one too? Thanks.

Thanks Bernd, the refactor looks good to me

**Review by marco-miller — APPROVED** [review]
LGTM now, thanks, I can `yarn` multiple times locally.

**Review by bhufmann — APPROVED** [review]
Looks good to me too. Thanks everybody involved.

**Pull request merged**